check a symbol for null before descending into its base type

brettfo opened the pull request:
When preparing quick info for attributes, there are corner cases where the symbol's containing type could be null (e.g., the type or one of its subtypes comes from an unreferenced assembly.)  The ultimate entry into this case was really in [AbstractSemanticQuickInfoProvider.cs](https://github.com/dotnet/roslyn/blob/8053b00c08ac33c44917d9f6eb20f6467ae7b55d/src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Providers/AbstractSemanticQuickInfoProvider.cs#L238) but in reviewing calls to `ITypeSymbolExtensions.IsAttribute()` there were at least two other locations that could potentially hit the same null ref, [here](https://github.com/dotnet/roslyn/blob/8053b00c08ac33c44917d9f6eb20f6467ae7b55d/src/Workspaces/VisualBasic/Portable/Simplification/VisualBasicSimplificationService.Expander.vb#L484) and [here](https://github.com/dotnet/roslyn/blob/8053b00c08ac33c44917d9f6eb20f6467ae7b55d/src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.Expander.cs#L429), so I though it would be better to fix it at the root rather than patch all of the call sites and hope no more were created, though I'm certainly open to discussion here.

I was unable to repro this manually and I couldn't construct a failing test without a lot of hackery, hence no added/updated unit test.

Fixes #2440.

brettfo commented:
Tagging @Pilchie @dpoeschl @davkean @rchande @jasonmalinowski @balajikris @basoundr @jmarolf as potential reviewers

rchande commented:
:+1:

balajikris commented:
LGTM.

jasonmalinowski commented on src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs:
```diff
@@ -352,16 +352,19 @@ public static IEnumerable<INamedTypeSymbol> GetContainingTypes(this ITypeSymbol
-                    b.ContainingNamespace.Name == "System" &&
-                    b.ContainingNamespace.ContainingNamespace != null &&
-                    b.ContainingNamespace.ContainingNamespace.IsGlobalNamespace)
+            if (symbol != null)
```
Do we want to continue down the pattern of allowing null to be passed to extension methods?

Pilchie replied:
I'd rather not for these cases.  I get it for IsKind/IsParentKind, but this seems like a different sort of thing.

jasonmalinowski replied:
Does ?. work on extension method invocations?

Pilchie replied:
Yes, why wouldn't it?

jasonmalinowski commented:
Tests?

brettfo commented:
Updated to null check at the call sites instead of in the extension method.

@jasonmalinowski, I checked to ensure that Elvis works on extension methods and as to the no tests question, I addressed it briefly in the initial comment, but when talking to @rchande about when he found this, we don't have a manual repro or a call stack and creating a scenario where a constructor symbol doesn't have a containing type is difficult and it felt contrived to create one just to test a null check.

Pilchie commented:
LGTM, but if you have a crashdump, you can probably extract the file that caused the null parented attribute.  The compiler team might be interested in it in case they don't expect it either.
